packages/patterns/test/types.test-d.ts

```diff
@@ -2,7 +2,7 @@ import type { Passable } from '@endo/pass-style';
 import { expectNotType, expectType } from 'tsd';
 import { isKey, isScalarKey } from '../src/keys/checkKey.js';
 import { M } from '../src/patterns/patternMatchers.js';
```

Comment:
Nit: not strictly in scope of this PR, but I believe you can import the runtime exports of the package from the root of the package too. Also I think it's legal to import from `@endo/patterns` to go through the package.json exports resolution mechanism, verifying that everything is correctly exported. (in this case the `package.json` is pretty straightforward tho)